### PR TITLE
CI: update k8s version and CAPIRELEASE version

### PR DESCRIPTION
**Pull request by Jaakko-Os**
Upgrade tests script update needed due to
https://github.com/metal3-io/metal3-dev-env/pull/501

### CODE REVIEW

File: scripts/feature_tests/upgrade/upgrade_common.sh [hunk 1]
```diff
@@ -5,11 +5,11 @@ export CLUSTER_NAME=${CLUSTER_NAME:-"test1"}
...
 export IMAGE_USERNAME=${IMAGE_USERNAME:-"metal3"}
-export KUBERNETES_VERSION=${KUBERNETES_VERSION:-"v1.18.0"}
-export KUBERNETES_BINARIES_VERSION=${KUBERNETES_BINARIES_VERSION:-"v1.18.0"}
-export UPGRADED_K8S_VERSION_1=${UPGRADED_K8S_VERSION_1:-"v1.18.1"}
-export UPGRADED_K8S_VERSION_2=${UPGRADED_K8S_VERSION_2:-"v1.18.2"}
-export UPGRADED_BINARY_VERSION=${UPGRADED_BINARY_VERSION:-"v1.18.1"}
+export KUBERNETES_VERSION=${KUBERNETES_VERSION:-"v1.18.8"}
+export KUBERNETES_BINARIES_VERSION=${KUBERNETES_BINARIES_VERSION:-"v1.18.8"}
+export UPGRADED_K8S_VERSION_1=${UPGRADED_K8S_VERSION_1:-"v1.18.9"}
+export UPGRADED_K8S_VERSION_2=${UPGRADED_K8S_VERSION_2:-"v1.19.0"}
```
Comment (maelk) [inline scripts/feature_tests/upgrade/upgrade_common.sh:11]: you should verify that this is supported afaik v0.3.9 was not supporting upgrades to 1.19 . But maybe 0.3.10 does ?

Comment (Jaakko-Os) [reply scripts/feature_tests/upgrade/upgrade_common.sh]: I stepped down one patch version, v1.18.7 to v1.18.9 now used

**Comment by Jaakko-Os** [comment]
> /test-upgrade-features

https://github.com/metal3-io/metal3-dev-env/pull/508 is to be merged before running CI jobs

File: scripts/feature_tests/upgrade/upgrade_common.sh [hunk 2]
```diff
@@ -5,21 +5,25 @@ export CLUSTER_NAME=${CLUSTER_NAME:-"test1"}
...
```
Comment (Xenwar) [inline scripts/feature_tests/upgrade/upgrade_common.sh:11]: set both to v1.18.9

File: scripts/feature_tests/upgrade/upgrade_common.sh [hunk 3]
```diff
@@ -471,5 +473,32 @@ function verify_kubernetes_version_upgrade() {
...
+}
+
+function point_to_target_cluster() {
+    export KUBECONFIG="${TARGET_KUBECONFIG_FILE}"
 
 }
+
+function point_to_management_cluster() {
+    unset KUBECONFIG
+    rm "${TARGET_KUBECONFIG_FILE}"
```
Comment (Xenwar) [inline scripts/feature_tests/upgrade/upgrade_common.sh:490]: you do not need this.

File: scripts/feature_tests/upgrade/upgrade_common.sh [hunk 4]
```diff
@@ -144,17 +148,17 @@ EOF
 }
 
 function manage_node_taints() {
-    kubectl get secrets "${CLUSTER_NAME}"-kubeconfig -n "${NAMESPACE}" -o json | \
-      jq -r '.data.value'| base64 -d > /tmp/kubeconfig-"${CLUSTER_NAME}".yaml
+    set_kubeconfig_towards_target_cluster
```
Comment (Xenwar) [inline scripts/feature_tests/upgrade/upgrade_common.sh:151]: Calling this function multiple times, specially after pivoting has an undesired result.
Eg. after pivoting the secret does not exist on the source cluster which results in an empty kubeconfig file

**Review by Xenwar — CHANGES_REQUESTED** [review]

File: scripts/feature_tests/upgrade/upgrade_common.sh (hunk 3 above)
Comment (Jaakko-Os) [reply scripts/feature_tests/upgrade/upgrade_common.sh]: removed

File: scripts/feature_tests/upgrade/upgrade_common.sh (hunk 2 above)
Comment (Jaakko-Os) [reply scripts/feature_tests/upgrade/upgrade_common.sh]: done

File: scripts/feature_tests/upgrade/upgrade_common.sh (hunk 4 above)
Comment (Jaakko-Os) [reply scripts/feature_tests/upgrade/upgrade_common.sh]: fixed

**Review by smoshiur1237 — COMMENTED** [review]
Looks good to me.

File: scripts/feature_tests/upgrade/upgrade_common.sh (hunk 1 above)
Comment (Jaakko-Os) [reply scripts/feature_tests/upgrade/upgrade_common.sh]: 1.18.9 used instead

**Comment by Jaakko-Os** [comment]
To overcome the verification challenges, this PR is split to smaller pieces.

**Pull request closed without merging**